Transalte-c generates invalid zig (unescaped characters)

### Zig Version

0.10.0-dev.3659+e5e6eb983

### Steps to Reproduce

- `asd.h` contains:

  ```c
  #define asd "ASD	"
  ```
  (the character after ASD is a tab)

- `asd.zig` contains:
  ```zig
  const std = @import("std");
  const c = @cImport({
      @cInclude("asd.h");
  });
  
  test{
      std.debug.print("{s}\n", .{c.asd});
  }
  ```
- run `zig test asd.zig -lc -I.`

### Expected Behavior

Compile successfully

### Actual Behavior

Compilation fails:
```
zig-cache/o/2ba70bf5d1fe5df47298c857c8beed71/cimport.zig:467:21: error: expected ';' after declaration
pub const asd = "ASD ";
                    ^~
zig-cache/o/2ba70bf5d1fe5df47298c857c8beed71/cimport.zig:467:23: note: invalid byte: ';'
pub const asd = "ASD ";
                      ^
asd.zig:2:15: error: C import failed: AnalysisFail
pub const c = @cImport({
              ^~~~~~~~
```

With `-fstage1` it compiles successfully, even though the offending tab character is still in the generated `cimport.zig` file.

--- a/src/translate_c.zig
+++ b/src/translate_c.zig
@@ -5799,7 +5799,7 @@ fn zigifyEscapeSequences(ctx: *Context, m: *MacroCtx) ![]const u8 {
         }
     }
     for (source) |c| {
-        if (c == '\\') {
+        if (c == '\\' or c == '\t') {
             break;
         }
     } else return source;
@@ -5876,6 +5876,13 @@ fn zigifyEscapeSequences(ctx: *Context, m: *MacroCtx) ![]const u8 {
                     state = .Start;
             },
             .Start => {
+                if (c == '\t') {
+                    bytes[i] = '\\';
+                    i += 1;
+                    bytes[i] = 't';
+                    i += 1;
+                    continue;
+                }
                 if (c == '\\') {
                     state = .Escape;
                 }

--- a/test/behavior/translate_c_macros.h
+++ b/test/behavior/translate_c_macros.h
@@ -50,3 +50,5 @@ typedef _Bool uintptr_t;
 #define CAST_TO_UINTPTR(X) (uintptr_t)(X)
 
 #define LARGE_INT 18446744073709550592
+
+#define EMBEDDED_TAB "hello	"

--- a/test/behavior/translate_c_macros.zig
+++ b/test/behavior/translate_c_macros.zig
@@ -2,6 +2,7 @@ const builtin = @import("builtin");
 const std = @import("std");
 const expect = std.testing.expect;
 const expectEqual = std.testing.expectEqual;
+const expectEqualStrings = std.testing.expectEqualStrings;
 
 const h = @cImport(@cInclude("behavior/translate_c_macros.h"));
 
@@ -122,4 +123,14 @@ test "large integer macro" {
     if (builtin.zig_backend == .stage2_aarch64) return error.SkipZigTest; // TODO
 
     try expectEqual(@as(c_ulonglong, 18446744073709550592), h.LARGE_INT);
+}
+
+test "string literal macro with embedded tab character" {
+    if (builtin.zig_backend == .stage2_wasm) return error.SkipZigTest; // TODO
+    if (builtin.zig_backend == .stage2_x86_64) return error.SkipZigTest; // TODO
+    if (builtin.zig_backend == .stage2_c) return error.SkipZigTest; // TODO
+    if (builtin.zig_backend == .stage2_arm) return error.SkipZigTest; // TODO
+    if (builtin.zig_backend == .stage2_aarch64) return error.SkipZigTest; // TODO
+
+    try expectEqualStrings("hello\t", h.EMBEDDED_TAB);
 }

